### ISSUE TITLE
Question: I want to develop application for student study planning

### ISSUE THREAD

**Issue by sameerdube9** [rubenwe/PlanningAI#3]
Hello @rubenwe 

I want to develop student study planning application where student will select course =>subjects => chapters => topics (each one has some difficulty level set by teacher and approximate time to complete study for that topic).

student will have option to select exam date and number of revision he/she needs to complete.

planning library should propose best plan considering holidays and other festival consideration and study time (per day in hours).

We also want to track the progress for student whether he/she is following plan or not (this can be done by mobile app which will report daily study completion)

if there is delay then subsequent changes to plan and other measures such as reducing revisions in study plan for the student.

**will this library help me achieve certain parts of above requirements?, if so, please guide me in right direction**

**Comment by rubenwe** [comment]
Hey @samdubey ,

I don't think Goal Oriented Action Planning will help you too much with that.
Honestly sounds like a problem that is solvable without a lot of "search".

I guess you'd have a fixed number of slots per day that can be allocated to revisions.
You'll need to fill (some) of those up. And how you distribute the lessons is really a matter of preference on the students part.

I personally learn better with a moderate amount of pressure - so an algorithm that puts most of the lessons somewhat close to the exam and bunches together learning sessions might be nice.
But there are others that benefit more for continuous learning and repetition. 
If you are building this you should have better insight into the domain and learning behaviours than I do. 

You could distribute via a weighted search algorithm (just like the A*-algo that is at the core of my planning) but I don't know if that is the best way to do this or if a simple brute force algo or something more advanced is the best way to go here.

**TLDR;** maybe, but I don't think it's a prefect fit to solve your problems.

**Comment by sameerdube9** [comment]
Thank you very much for your answer
